### PR TITLE
Add a very basic logger shell service

### PR DESCRIPTION
**Pull request by nashif**
This service will store all syslog messages into a ring buffer and display
them in the shell environment on demand. Very useful when you have a shell
application where logging is enabled, making usage of the shell very
annoying.
Now you can look at the syslog whenever you desire. Buffer size can be set
for heavy logging.

Also fix object tracing for ring buffers and fix logger hook to handle system
messages. (init early on)

### CODE REVIEW

File: subsys/shell/modules/logger_service.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	u16_t type;
+	u8_t val;
+
+	key = irq_lock();
+	ret = sys_ring_buf_get(&log_cbuffer.ring_buffer, &type, &val,
+			       (u32_t *)data, &size32);
+	irq_unlock(key);
+
+	if (ret == 0) {
+		printk("%s", data);
```
Comment (lpereira) [inline subsys/shell/modules/logger_service.c]: Data is potentially not null-terminated; printing without considering `size32` will most likely lead to crashes.

File: subsys/shell/modules/logger_service.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+void vlog_cbuf_put(const char *format, va_list args)
+{
+	char buf[512];
+	int buf_size = 0;
+
+	buf_size += vsnprintf(&buf[buf_size], sizeof(buf), format, args);
+	logger_put(&log_cbuffer, buf, buf_size);
+}
+
+void log_cbuf_put(const char *format, ...)
```
Comment (lpereira) [inline subsys/shell/modules/logger_service.c]: Do we need these functions to be exported? Can't them be all static?

File: subsys/shell/modules/logger_service.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	ret = sys_ring_buf_put(&logger->ring_buffer, 0, 0,
+			       (u32_t *)data, size32);
+	irq_unlock(key);
+
+	return ret;
+}
+
+void vlog_cbuf_put(const char *format, va_list args)
+{
+	char buf[512];
```
Comment (lpereira) [inline subsys/shell/modules/logger_service.c]: Why 512? Shouldn't this be in function of CONFIG_LOGGER_BUFFER_SIZE?

**Comment by nashif** [comment]
thinking about it this more, the logger backend should not be part of the shell, it should be part of syslog, the code was a copy of a sample and has some other issues beside the above.

**Pull request closed without merging**